src/Functions/ztest.cpp

```diff
@@ -0,0 +1,171 @@
+#include <Columns/ColumnNullable.h>
+#include <Columns/ColumnTuple.h>
+#include <Columns/ColumnString.h>
+#include <Columns/ColumnsNumber.h>
+#include <DataTypes/DataTypeTuple.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <Functions/IFunction.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionHelpers.h>
+#include <boost/math/distributions/normal.hpp>
+
+
+namespace DB
+{
+
+    namespace ErrorCodes
+    {
+        extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+    }
+
+
+    class FunctionTwoSampleProportionsZTest : public IFunction
+    {
+    public:
+        static constexpr auto POOLED = "pooled";
+        static constexpr auto UNPOOLED  = "unpooled";
+
+        static constexpr auto name = "proportionsZTest";
+
+        static FunctionPtr create(ContextPtr)
+        {
+            return std::make_shared<FunctionTwoSampleProportionsZTest>();
+        }
+
+        String getName() const override
+        {
+            return name;
+        }
+
+        size_t getNumberOfArguments() const override { return 6; }
+
+        bool useDefaultImplementationForNulls() const override { return false; }
+        bool useDefaultImplementationForConstants() const override { return true; }
+        bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return false; }
+
+        static DataTypePtr getReturnType()
+        {
+            DataTypes types
+            {
+                std::make_shared<DataTypeNumber<Float64>>(),
+                std::make_shared<DataTypeNumber<Float64>>(),
+                std::make_shared<DataTypeNumber<Float64>>(),
+                std::make_shared<DataTypeNumber<Float64>>(),
+            };
+
+            Strings names
+            {
+                "z_statistic",
+                "p_value",
+                "confidence_interval_low",
+                "confidence_interval_high"
+            };
+
+            return std::make_shared<DataTypeTuple>(
+                std::move(types),
+                std::move(names)
+            );
+        }
+
+        DataTypePtr getReturnTypeImpl(const DataTypes & /*arguments*/) const override
+        {
+            return getReturnType();
```

Comment:
Almost always we need to validate arguments here, to safely cast columns to right types in executeImpl.

Reply:
I fixed it.